src/TodoApp/TodoApp.test.jsx

```diff
@@ -0,0 +1,107 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { TodoApp } from './TodoApp';
+
+describe('Test on <TodoApp />', () => {
+	const getElements = (screen) => {
+		const input = screen.getByLabelText('input-task');
+		const button = screen.getByLabelText('add-task');
+		const reset = screen.getByLabelText('reset-todo-list');
+		const remaining = screen.getByLabelText('remaining-tasks');
+
+		fireEvent.click(reset);
+
+		return {
+			input,
+			button,
+			reset,
+			remaining,
+		};
+	};
+
+	test('should can add 2 tasks with diferent value', () => {
+		render(<TodoApp />);
+
+		const { input, button } = getElements(screen);
+
+		fireEvent.change(input, { target: { value: 'task 1' } });
+		fireEvent.click(button);
+		fireEvent.change(input, { target: { value: 'task 2' } });
+		fireEvent.click(button);
+
+		const tasks = screen.getAllByLabelText('task-item');
+
+		expect(tasks.length).toBe(2);
+	});
+
+	test("should can't add 2 tasks with same value", () => {
+		render(<TodoApp />);
+
+		const { input, button } = getElements(screen);
+
+		fireEvent.change(input, { target: { value: 'task 1' } });
+		fireEvent.click(button);
+		fireEvent.change(input, { target: { value: 'task 1' } });
+		fireEvent.click(button);
+
+		const tasks = screen.getAllByLabelText('task-item');
+
+		expect(tasks.length).toBe(1);
+	});
+
+	test('should can reset todo list', () => {
```

Comment:
You should try to group your tests according to the different behaviors by using describe, which can be nested, example:

```
describe('Circle class', function() {
  describe('area is calculated when', function() {
    it('sets the radius', function() { ... });
    it('sets the diameter', function() { ... });
    it('sets the circumference', function() { ... });
  });
});
```

In this case you can group the tests like:

```
describe('<TodoApp />', () => {
  describe('adding new items', () => { ... });
  describe('clearing all the items', () => { ... });
  describe('task completion', () => { ... });
});
```

Use describe as a tool to group related behaviours.

Reply:
On bigger applications, you definitely don't want to add just tests for the main `App` component, but you want to build tests for each component in different files. Smaller test suites are good for maintainable code.